### DISCUSSION TITLE
API?

### DISCUSSION THREAD

**Discussion by j3ko** [signag/raspi-cam-srv#34] [Ideas]
Nice project!  Out of all the software I have tested, this is the only one that doesn't completely lock up my rpi zero.  Just wondering, is there an API on the roadmap for this? This would be perfect if I could hook the software up to node-red to control it remotely.  Ideally, functionality like: turn motion on/off, take a snapshot, make a 30s video, query status, etc.

I see all the functionality is already there in web methods (/stop_triggered_capture, /stop_triggered_capture) but to call them, authentication is needed.  Is there already a way to bypass authentication?  Or maybe an API token can be added for each user to bypass auth?

Also, it would be nice if motion notifications could also post to a webhook.  But that is minor since node-red can emulate an smtp server.

**Comment by signag** [comment]
Hi,
thanks for your post and suggestions.

You are the first requesting this kind of API authentication. 

Actually, authentication is based on the Flask session which uses cookies and lives as long as a browser session is active.
I could think of an additional authentication solution where
1. an authorized user generates a token which is only visible to him while it is being generated
2. this token can be used in a service call for authentication and bypass of session authentication
Would this cover your needs?

I will also think about support of notification to webhooks.

**Comment by j3ko** [comment]
Yes! that would be perfect.  Calling webhooks would be super nice too.  Being able to communicate with the application via API would make that app very useful because you can automate the functionality.  Thank you for taking the suggestion.

**Comment by signag** [comment]
Hi,
finally, I found some time to implement an API for raspiCamSrv.
It is now available in V2.11 (See [Release Notes](https://github.com/signag/raspi-cam-srv/blob/main/docs/ReleaseNotes.md#v2110))

**Reply by j3ko** [reply]
awesome! thank you!
